staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -117,12 +190,223 @@ func (a *ApplySet) addLabels(objects []*resource.Info) error {
 	return nil
 }
 
-func (p *ApplySetParentRef) IsNamespaced() bool {
-	return p.RESTMapping.Scope.Name() == meta.RESTScopeNameNamespace
+// MarkObjectVisited keeps track of the applied objects, so that we can update the list of in-use namespaces
+// and resouce kinds, and so that we can prune objects not applied.
+func (a *ApplySet) MarkObjectVisited(resource *meta.RESTMapping, namespace string) {
+	a.updatedResources[resource.Resource] = resource
+	if namespace != "" {
+		a.updatedNamespaces.Insert(namespace)
+	}
+}
+
+func (a *ApplySet) FetchParent() error {
+	helper := resource.NewHelper(a.client, a.parent.RESTMapping)
+	obj, err := helper.Get(a.parent.Namespace, a.parent.Name)
+	if errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to fetch ApplySet parent object %q from server: %w", a.ParentRef(), err)
+	} else if obj == nil {
+		return fmt.Errorf("failed to fetch ApplySet parent object %q from server", a.ParentRef())
+	}
+
+	labels, annotations := getLabelsAndAnnotations(obj)
+
+	toolAnnotation, hasToolAnno := annotations[ApplySetToolingAnnotation]
+	if !hasToolAnno {
+		return fmt.Errorf("ApplySet parent object %q already exists and is missing required annotation %q", a.ParentRef(), ApplySetToolingAnnotation)
+	}
+	if managedBy := toolingBaseName(toolAnnotation); managedBy != a.toolingID.name {
+		return fmt.Errorf("ApplySet parent object %q already exists and is managed by tooling %s instead of %s", a.ParentRef(), managedBy, a.toolingID.name)
```

Comment:
Super nit - I like to use "tooling %q instead of %q" because I think the quotes makes it clearer that we're just referring to the string values rather than knowing that helm is managing something, also because if there was a whitespace difference (which I don't think there can be here) it would be more obvious